x-pack/plugins/observability_solution/entities_data_access/server/services/index_pattern_service.test.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { savedObjectsClientMock } from '@kbn/core-saved-objects-api-server-mocks';
+import { createIndexPatternService } from './index_pattern_service';
+
+describe('index_pattern_service', () => {
+  const indexPatternService = createIndexPatternService();
+
+  describe('#indexPattern', () => {
+    it('returns wide index pattern when all datasets and version specified', () => {
+      let pattern = indexPatternService.indexPattern();
+      expect(pattern).toEqual('.entities.*.*.*');
+
+      pattern = indexPatternService.indexPattern({ datasets: 'all', schemaVersion: 'all' });
+      expect(pattern).toEqual('.entities.*.*.*');
```

Comment:
I wonder if there is a performance downside to doing `.entities.*.*.*` compared to doing `.entities*`?
It might be a pre-optimization thing, but perhaps we could replace `.entities.*.*.*` with ``.entities*` if we find it in the list of patterns before the function returns?